mlir/test/Integration/Dialect/Linalg/CPU/ArmSME/lit.local.cfg

```diff
@@ -4,6 +4,10 @@ import sys
 if not config.mlir_run_arm_sme_tests:
     config.unsupported = True
 
+# With no hardware available, ArmSME tests require emulation.
+if not config.arm_emulator_executable:
+    config.unsupported = True
+
```

Comment:
I would rather have this be an error: we shouldn't allow `-DMLIR_RUN_ARM_SME_TESTS=On` without the emulator. 
Here the user will see  `ninja check-mlir`  complete successfully but ignoring these tests despite the CMake flag enabling them.

Reply:
Agreed, thanks! I've just updated this PR to use CMake instead.